### PR TITLE
Load balancer settings

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/652

### CODE REVIEW

**Comment by benishak** [comment]
great, thanks! 

could you also make these settings available
- max_requests_per_connection
- connect_timeout
- upstream_connection_options
- close_connections_on_host_health_failure
- health_checks (or are you auto taking this config from ReadinessProbe and LivenessProbe?) 

https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cds.proto

File: projects/gloo/pkg/plugins/loadbalancer/plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		upstreamSpec.LoadBalancerConfig = &v1.LoadBalancerConfig{
+			UpdateMergeWindow: &t,
+		}
+		err := plugin.ProcessUpstream(params, upstream, out)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(out.CommonLbConfig.UpdateMergeWindow.Seconds).To(BeEquivalentTo(1))
+		Expect(out.CommonLbConfig.UpdateMergeWindow.Nanos).To(BeEquivalentTo(0))
+	})
+
+	It("should set UpdateMergeWindow", func() {
```
Comment (mitchdraft) [inline projects/gloo/pkg/plugins/loadbalancer/plugin_test.go]: this is a duplicate of the above

**Comment by yuval-k** [comment]
let me see about that - It should be pretty easy to add those

**Comment by yuval-k** [comment]
@benishak note that `close_connections_on_host_health_failure` is documented to be `... only supported for connections created by tcp_proxy.`

**Comment by benishak** [comment]
ok then not need for it

**Comment by benishak** [comment]
@yuval-k any ETA for that? i will open an issue for reference

**Pull request merged**